### ISSUE TITLE
[prefer-optional-chain] false positive if chain is returned and return type is boolean

### ISSUE THREAD

**Issue by robstoll** [typescript-eslint/typescript-eslint#4490]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [ ] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-optional-chain": "error"
  }
}
```
in an Angular project
```TS
 hasRequiredError(form: FormGroup, formFieldName: string): boolean {
    const field = form.get(formFieldName);
    return field != null && field.hasError('required');
    // ^ @typescript-eslint/prefer-optional-chain
  }
```

**Expected Result**
don't warn in this case as `field?.hasError('required')` is no longer a boolean and requires something like `!!field?.hasError('required')` or `field?.hasError('required') ?? false` which are both less readable than the initial solution IMO.

I would still warn in case the expression is not returned but used in an if statement or the like.

**Actual Result**
complains

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^4.33.0`|
| `@typescript-eslint/parser`        | `^4.33.0` |
| `TypeScript`                       | `~4.2.4` |
| `ESLint`                           | `^7.32.0` |
| `node`                             | `v14.16.0` |

**Comment by bradzacher** [comment]
Duplicate of #2963

The rule doesn't use type information, so this is expected.

**Comment by robstoll** [comment]
thanks for the fast answer
